### PR TITLE
[Quant] Add fused LinearTanh module for onednn backend

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89188
* __->__ #88923
* #88879
* #88668

**Summary**
This PR adds fused `QLinearTanh` module for onednn backend, which will be used for int8 inference with onednn backend. Cannot call this module with other quantization backends otherwise an error is thrown.

**Test plan**
python test_quantization.py TestStaticQuantizedModule

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by jerryzh168 — COMMENTED** [review]
looks good overall, but probably need some update after the previous PRs are updated I think

**Review by jgong5 — APPROVED** [review]
Not sure if we need to rename the files to something like linear_activation from linear_relu since we added linear+tanh into it.

**Comment by Xia-Weiwen** [comment]
> Not sure if we need to rename the files to something like linear_activation from linear_relu since we added linear+tanh into it.

Good idea. Thanks. I will do that after clarifications from Jerry about PT 2.0.

**Comment by Xia-Weiwen** [comment]
Hi @jerryzh168. Is it ok to land this? Thanks

**Comment by Xia-Weiwen** [comment]
Hi @jerryzh168 Do you have more comments on this PR? Thanks!

File: torch/nn/intrinsic/__init__.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 from torch.ao.nn.intrinsic import BNReLU3d
 from torch.ao.nn.intrinsic import LinearBn1d
 from torch.ao.nn.intrinsic import LinearLeakyReLU
+from torch.ao.nn.intrinsic import LinearTanh
```
Comment (jerryzh168) [inline torch/nn/intrinsic/__init__.py:15]: please revert these, we don't want to add new things to torch/nn/intrinsic folder, this is a folder that we plan to deprecate in the future

**Review by jerryzh168 — CHANGES_REQUESTED** [review]
please revert changes to torch/nn/intrinsic folder

File: torch/nn/intrinsic/__init__.py (hunk 1 above)
Comment (Xia-Weiwen) [reply torch/nn/intrinsic/__init__.py]: Ok they are reverted.

**Comment by Xia-Weiwen** [comment]
> please revert changes to torch/nn/intrinsic folder

Ok they are reverted.

File: test/allowlist_for_publicAPI.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "torch.nn.intrinsic.quantized.modules": "torch.ao.nn.intrinsic.quantized.modules",
     "torch.nn.intrinsic.quantized.modules.bn_relu": "torch.ao.nn.intrinsic.quantized.modules.bn_relu",
     "torch.nn.intrinsic.quantized.modules.conv_relu": "torch.ao.nn.intrinsic.quantized.modules.conv_relu",
-    "torch.nn.intrinsic.quantized.modules.linear_relu": "torch.ao.nn.intrinsic.quantized.modules.linear_relu",
+    "torch.nn.intrinsic.quantized.modules.linear_activation": "torch.ao.nn.intrinsic.quantized.modules.linear_activation",
```
Comment (jerryzh168) [inline test/allowlist_for_publicAPI.json:15]: is this change still valid? we don't have `torch.nn.intrinsic.quantized.modules.linear_activation` now

**Review by jerryzh168 — APPROVED** [review]

File: test/allowlist_for_publicAPI.json (hunk 2 above)
Comment (Xia-Weiwen) [reply test/allowlist_for_publicAPI.json]: It's still needed whereas the file name id changed:
```
"torch.nn.intrinsic.quantized.modules.linear_relu": "torch.ao.nn.intrinsic.quantized.modules.linear_activation",
```

Comment (Xia-Weiwen) [reply test/allowlist_for_publicAPI.json]: I reverted the file name change due to CI checks failure.

**Comment by Xia-Weiwen** [comment]
> > > Not sure if we need to rename the files to something like linear_activation from linear_relu since we added linear+tanh into it.
> > 
> > 
> > Good idea. Thanks. I will do that after clarifications from Jerry about PT 2.0.
> 
> I have renamed the file to `linear_activation.py`

I reverted the change due to CI check failure.

**Pull request closed without merging**